lib/module.js

```diff
@@ -90,8 +94,13 @@ function tryPackage(requestPath, exts) {
   if (!pkg) return false;
 
   var filename = path.resolve(requestPath, pkg);
-  return tryFile(filename) || tryExtensions(filename, exts) ||
-         tryExtensions(path.resolve(filename, 'index'), exts);
+  var result = tryFile(filename) || tryExtensions(filename, exts) ||
+      tryExtensions(path.resolve(filename, 'index'), exts);
+  if (!result) {
+    throw new Error(
+        'Could not resolve "main" in ' + getPackageJsonPath(requestPath));
```

Comment:
hehe, github diff collapsing - making all code look like the same function! :tada: